sdk/keyvault/Azure.Security.KeyVault.Keys/tests/KeyClientLiveTests.cs

```diff
@@ -19,6 +19,19 @@ public KeyClientLiveTests(bool isAsync) : base(isAsync)
         {
         }
 
+        [SetUp]
+        public void ClearChallengeCacheforRecord()
+        {
+            // in record mode we reset the challenge cache before each test so that the challenge call
+            // is always made.  This allows tests to be replayed independently and in any order
+            if (this.Mode == RecordedTestMode.Record || this.Mode == RecordedTestMode.Playback)
+            {
+                this.Client = this.GetClient();
+
+                ChallengeBasedAuthenticationPolicy.AuthenticationChallenge.ClearCache();
```

Comment:
I wonder if it's worth to have this as an internal client option (something like IsGlobalChallengeCacheEnabled) and set it in `GetClient`, not sure if it's cleaner though.